### PR TITLE
[bitnami/valkey-cluster] Add valkey cluster chart

### PR DESCRIPTION
**Pull request by rafariossaa**
### Description of the change

Adds chart for `valkey-cluster`

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/valkey-cluster/templates/svc-cluster-external-access.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+{{- $nodesCount := .Values.cluster.nodes | int }}
+{{- $root := . }}
+
+{{- range $i, $e := until $nodesCount }}
+{{- $targetPod := printf "%s-%d" (printf "%s" $fullName) $i }}
+{{- $_ := set $ "targetPod" $targetPod }}
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ template "common.names.fullname" $ }}-{{ $i }}-svc
```
Comment (fmulero) [inline bitnami/valkey-cluster/templates/svc-cluster-external-access.yaml:17]: Be careful with the names

File: bitnami/valkey-cluster/templates/valkey-role.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  {{- if .Values.commonAnnotations }}
+  annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
+  {{- end }}
+rules:
+  {{- if and (include "common.capabilities.psp.supported" .) .Values.podSecurityPolicy.create }}
+  - apiGroups: ['{{ template "podSecurityPolicy.apiGroup" . }}']
+    resources: ['podsecuritypolicies']
+    verbs: ['use']
+    resourceNames: [{{ template "common.names.fullname" . }}]
+{{- end -}}
```
Comment (fmulero) [inline bitnami/valkey-cluster/templates/valkey-role.yaml:22]: I think we could remove it

File: bitnami/valkey-cluster/templates/valkey-statefulset.yaml [hunk 3]
```diff
@@ -0,0 +1,510 @@
...
+        {{- include "common.tplvalues.render" (dict "value" .Values.valkey.podAnnotations "context" $) | nindent 8 }}
+        {{- end }}
+        {{- if and .Values.metrics.enabled .Values.metrics.podAnnotations }}
+        {{- include "common.tplvalues.render" (dict "value" .Values.metrics.podAnnotations "context" $) | nindent 8 }}
+        {{- end }}
+    spec:
+      hostNetwork: {{ .Values.valkey.hostNetwork }}
+      {{- if semverCompare ">= 1.13" (include "common.capabilities.kubeVersion" .) }}
+      enableServiceLinks: false
+      {{- end }}
```
Comment (fmulero) [inline bitnami/valkey-cluster/templates/valkey-statefulset.yaml:48]: According to our README file, we support versions 1.23+. I think this condition is not needed and we could allow users change this value

File: bitnami/valkey-cluster/templates/headless-svc.yaml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  type: ClusterIP
+  clusterIP: None
+  publishNotReadyAddresses: true
+  ports:
+    - name: tcp-redis
+      port: {{ .Values.valkey.containerPorts.valkey }}
+      targetPort: tcp-redis
+    - name: tcp-redis-bus
+      port: {{ .Values.valkey.containerPorts.bus }}
+      targetPort: tcp-redis-bus
```
Comment (fmulero) [inline bitnami/valkey-cluster/templates/headless-svc.yaml:26]: mmm, the port name in the `valkey.containerPorts` value was changed so I think we should do the same for this `tcp-redis` and `tcp-redis-bus` port names, What do you think?

File: bitnami/valkey-cluster/Chart.yaml [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+# SPDX-License-Identifier: APACHE-2.0
+
+annotations:
+  category: Database
+  licenses: Apache-2.0
+  images: |
+    - name: os-shell
+      image: docker.io/bitnami/os-shell:12-debian-12-r22
+    - name: valkey-cluster
+      image: docker.io/bitnami/redis-cluster:7.2.5-debian-12-r0
```
Comment (fmulero) [inline bitnami/valkey-cluster/Chart.yaml:11]: redis-cluster?

File: bitnami/valkey-cluster/Chart.yaml [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+keywords:
+- valkey
+- keyvalue
+- database
+maintainers:
+- name: Broadcom, Inc. All Rights Reserved.
+  url: https://github.com/bitnami/charts
+name: valkey-cluster
+sources:
+- https://github.com/bitnami/charts/tree/main/bitnami/valkey-cluster
```
Comment (fmulero) [inline bitnami/valkey-cluster/Chart.yaml:34]:
```suggestion
- https://github.com/bitnami/charts/tree/main/bitnami/valkey-cluster
- https://github.com/bitnami/containers/tree/main/bitnami/vakey-cluster
```

**Review by fmulero — COMMENTED** [review]
Thanks a lot @rafariossaa !!

I left some comments, could you give it a glance?

File: bitnami/valkey-cluster/templates/headless-svc.yaml (hunk 4 above)
Comment (fmulero) [reply bitnami/valkey-cluster/templates/headless-svc.yaml:26]: Ok no problem, I've just seen [the same in valkey](https://github.com/bitnami/charts/blob/5d703bad49025ab95654a09ecacef304de7492bd/bitnami/valkey/templates/headless-svc.yaml#L28)

Comment (rafariossaa) [reply bitnami/valkey-cluster/templates/headless-svc.yaml:26]: I am keeping `tcp-redis` for consistency with `valkey`chart.

File: bitnami/valkey-cluster/templates/psp.yaml [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+  hostNetwork: false
+  hostPID: false
+  privileged: false
+  readOnlyRootFilesystem: false
+  requiredDropCapabilities:
+    - ALL
+  runAsUser:
+    rule: 'MustRunAs'
+    ranges:
+      - min: {{ .Values.podSecurityContext.runAsUser }}
```
[the remark this answers is not in the document]
Comment (rafariossaa) [reply bitnami/valkey-cluster/templates/psp.yaml]: I removed the object

**Review by fmulero — APPROVED** [review]
LGTM

File: .vib/valkey-cluster/ginkgo/go.mod [hunk 8]
```diff
@@ -0,0 +1,57 @@
+module test-redis-chart
```
Comment (jotamartos) [inline .vib/valkey-cluster/ginkgo/go.mod:1]:
```suggestion
module test-valkey-cluster-chart
```

**Review by jotamartos — APPROVED** [review]
Minor suggestion, the PR LGTM

File: .vib/valkey-cluster/ginkgo/go.mod (hunk 8 above)
Comment (rafariossaa) [reply .vib/valkey-cluster/ginkgo/go.mod:1]: I update this to `test-valkey-chart` the same way as `redis` and `redis-cluster` that share `test-redis-chart`

**Review by rafariossaa — COMMENTED** [review]
Updated

**Review by fmulero — APPROVED** [review]
LGTM

**Pull request merged**